pkg/pool-stable/contracts/StablePool.sol

```diff
@@ -116,19 +116,6 @@ contract StablePool is IStablePool, BalancerPoolToken, BasePoolAuthentication, P
     /// @notice Cannot stop an amplification update before it starts.
     error AmpUpdateNotStarted();
 
-    /**
-     * @notice Allow the swap manager to change the amplification parameter.
-     * @dev Unlike the swap fee percentage setting permission, this is non-exclusive.
-     */
-    modifier authenticateByRole() {
-        // Allow if this is the swapFeeManager.
-        if (msg.sender != _vault.getPoolRoleAccounts(address(this)).swapFeeManager) {
-            // Otherwise, defer to governance.
-            _authenticateCaller();
-        }
-        _;
-    }
```

Comment:
```
modifier authenticateByExclusiveRole() {
    address roleAddress = _vault.getPoolRoleAccounts(address(this)).swapFeeManager;

    if (roleAddress == address(0)) {
        // Defer to governance if no role assigned.
        _authenticateCaller();
    } else if (msg.sender != roleAddress) {
        revert SenderNotAllowed();
    }
    _;
}
```

Reply:
I would just follow the pattern established in `VaultAdmin`, and make the minimal change of replacing `authenticateByRole` with `authenticateByExclusiveRole`. It's specific to StablePool, so consider leaving it here vs. putting something up in BasePool that is swapFee-specific.

Reply:
Ahh... forgot you're sharing it with the MEV hook. Still, I'd say use standard terminology, and maybe define this up in BasePool, and have the local modifiers call it. That would be cleanest.

```
function _ensureAuthenticatedByExclusiveRole(address pool, address roleAddress) private view {
    if (roleAddress == address(0)) {
        // Defer to governance if no role assigned.
        _ensureAuthenticated(pool);
    } else if (msg.sender != roleAddress) {
        revert SenderNotAllowed();
    }
}
```

Reply:
@EndymionJkb mev hook is not a pool, so it's slightly different. I think the current change is fine (and in this case we always use swap fee manager, so it's fine to have it in the name of the modifier)